pyomo/core/base/reference.py

```diff
@@ -514,17 +517,34 @@ def Reference(reference, ctype=_NotSpecified):
               4 :     1 :    10 :  None : False : False :  Reals
 
     """
+    is_compdata = False
     if isinstance(reference, _IndexedComponent_slice):
         pass
     elif isinstance(reference, Component):
         reference = reference[...]
+    elif isinstance(reference, ComponentData):
+        index = reference.index()
+        parent = reference.parent_component()
+        if not isinstance(index, tuple):
+            index = (index,)
+        fixed = {i: val for i, val in enumerate(index)}
+        sliced = {}
+        ellipsis = None
+        reference = _IndexedComponent_slice(parent, fixed, sliced, ellipsis)
+        is_compdata = True
```

Comment:
@jsiirola I can create a slice at a singleton, but it doesn't behave as (I) expected with ```_ReferenceDict``` and ```_ReferenceSet```, so here I bypass these with the ```is_compdata``` flag and create my own ```_dict``` and ```index```. The result is accessed something like
```
m.r = Reference(m.v['a'])
m.r['a'].set_value(10)
```
This still creates some non-intuitive behavior though as the reference must be accessed through the index of the lowest-level container. E.g. for an ```IndexedVar``` in an ```IndexedBlock```
```
m.r2 = Reference(m.b[1].v1['b'])
m.r2['b'].set_value(10)
```
instead of
```
m.r2[1, 'b'].set_value(10)
```
as one would expect for a standard reference-to-slice:
```
m.r3 = Reference(m.b[:].v1[:])
m.r3[1, 'b'].set_value(10)
```
Not entirely sure how to fix this or if it's worth fixing. This still seems like a pretty superficial solution.